pkg/virt-controller/services/template.go

```diff
@@ -148,3 +167,63 @@ func NewTemplateService(launcherImage string, migratorImage string) (TemplateSer
 	}
 	return &svc, nil
 }
+
+// ComputeTotalMemoryRequirements computes the estimation of total
+// memory needed for the domain to operate properly.
+// This includes the memory needed for the guest and memory
+// for Qemu and OS overhead.
+//
+// The return values are requested memory and limit memory quantities
+//
+// Note: This is the best estimation we were able to come up with
+//       and is still not 100% accurate
+func computeTotalMemoryRequirements(vm *v1.VM) (*resource.Quantity, *resource.Quantity, error) {
+	domain := vm.Spec.Domain
+
+	// TODO this is pretty naive and does not work reliably, the libvirt units are different from quantity suffixes
+	memory, err := resource.ParseQuantity(fmt.Sprintf("%d%s", vm.Spec.Domain.Memory.Value, vm.Spec.Domain.Memory.Unit))
+	if err != nil {
+	  return nil, nil, err
+	}
+
+	limit, err := resource.ParseQuantity(fmt.Sprintf("%d%s", vm.Spec.Domain.MaxMemory.Value, vm.Spec.Domain.MaxMemory.Unit))
```

Comment:
Is there a reason why we don't remove the Unit field from Memory/MaxMemory and use the k8s unit representations?

We can use the k8s representation in the VM Spec and map that to the libvirt domain xml? That will make this easier and more reliable when mapping these units to the POD resource limits. It's also easier to read when viewing the VM Spec IMO.

Reply:
Yep, I want to do that, I just got a bit lost in the mappers and reflection.. Roman explained the mechanics there to me today so I might be able to update the mapper in virt-handler to use k8s Quantities.